test/integration/testpaths/testpath_snapshot_limits.py

```diff
@@ -0,0 +1,355 @@
+            cls.service_offering = ServiceOffering.create(
+                cls.apiclient,
+                cls.testdata["service_offering"],
+            )
+            cls._cleanup.append(cls.service_offering)
+
+            cls.disk_offering = DiskOffering.create(
+                cls.apiclient,
+                cls.testdata["disk_offering"],
+            )
+
+            cls._cleanup.append(cls.disk_offering)
+
+            cls.vm = VirtualMachine.create(
+                cls.userapiclient,
+                cls.testdata["small"],
+                templateid=cls.template.id,
+                accountid=cls.account.name,
+                domainid=cls.account.domainid,
+                serviceofferingid=cls.service_offering.id,
+                zoneid=cls.zone.id,
+                mode=cls.zone.networktype
+            )
+
+        except Exception as e:
+            cls.tearDownClass()
+            raise e
+        return
+
+    @classmethod
+    def tearDownClass(cls):
+        try:
+            cleanup_resources(cls.apiclient, cls._cleanup)
+        except Exception as e:
+            raise Exception("Warning: Exception during cleanup : %s" % e)
+
+    def setUp(self):
+        self.apiclient = self.testClient.getApiClient()
+        self.dbclient = self.testClient.getDbConnection()
+        self.cleanup = []
+
+    def tearDown(self):
+        try:
+            cleanup_resources(self.apiclient, self.cleanup)
+        except Exception as e:
+            raise Exception("Warning: Exception during cleanup : %s" % e)
+        return
+
+    @attr(tags=["basic", "advanced"])
+    def test_01_storage_snapshots_limits(self):
+        """ Storage and Snapshot Limit
+            1.   Create Snapshot of ROOT disk.
+            2.   Verify the Secondary Storage value
+                 is increased by the size of snapshot.
+            3.   Delete Snaphshot.
+            4.   Verify the Secondary
+                 Storage value is decreased by the size of snapshot.
+            5.   Set the Snapshot limit of Account.
+            6.   Create Snasphots till limit is reached.
+            7.   Create Snapshot of ROOT Volume.
+                 Creation should fail.
+            8.   Delete few Snapshots.
+            9.   Create Snapshot again.
+                 Creation should succeed.
+        """
+
+        # Get ROOT Volume
+        root_volumes_list = Volume.list(
+            self.userapiclient,
+            virtualmachineid=self.vm.id,
+            type='ROOT',
+            listall=True
+        )
+
+        self.assertEqual(
+            isinstance(
+                root_volumes_list,
+                list),
+            True,
+            "Check listVolumes for ROOT Disk returns a valid list")
+
+        self.assertNotEqual(len(root_volumes_list),
+                            0,
+                            "Check listVolumes response for ROOT Disk")
+
+        root_volume = root_volumes_list[0]
+
+        data_volume_created = Volume.create(
```

Comment:
Are you sure the data volume is deleted once test case execution completes? Although VM is deleted, volume will remain in the setup. It needs to be added to cleanup list separately. Please verify.